openpower/patches/p8dtu-patches/hostboot-p8/0012-Set-OPAL_MODEL-to-p8dtu1u-or-p8dtu2u-for-1U-and-2U-s.patch

```diff
@@ -0,0 +1,118 @@
+From 1e6bb450cf94fc3ade70f82703e6c57a30bbebaa Mon Sep 17 00:00:00 2001
+From: Jim Yuan <jim.yuan@supermicro.com>
+Date: Wed, 10 Aug 2016 10:41:32 -0700
+Subject: [PATCH 12/14] Set OPAL_MODEL to p8dtu1u or p8dtu2u for 1U and 2U
+ system.
+
+Signed-off-by: Jim Yuan <jim.yuan@supermicro.com>
+---
+ src/usr/hwpf/hwp/start_payload/start_payload.C     | 66 +++++++++++++++++++++-
+ .../common/xmltohb/attribute_types_hb.xml          |  1 +
+ 2 files changed, 66 insertions(+), 1 deletion(-)
+
+diff --git a/src/usr/hwpf/hwp/start_payload/start_payload.C b/src/usr/hwpf/hwp/start_payload/start_payload.C
+index ad155f19e..80e62a0b6 100644
+--- a/src/usr/hwpf/hwp/start_payload/start_payload.C
++++ b/src/usr/hwpf/hwp/start_payload/start_payload.C
+@@ -89,6 +89,8 @@
+ #include    <config.h>
+ #include    <ipmi/ipmiwatchdog.H>
+ #include    <ipmi/ipmiif.H> //jim
++#include	<string.h> //jim
++
+ #include    <vpd/vpd_if.H>
+ 
+ #include    <hwpf/hwpf_reasoncodes.H>
+@@ -487,6 +489,67 @@ void setAPSSGainOffsetFromPWSInfo(void)
+ 	assert(0);
+ 	}
+ }
++
++
++void setOpalMode (void)
++{
++
++ uint8_t riser_id = 0 ;
++ errlHndl_t l_err = NULL; 
++ size_t len = 4;
++ 
++ //create request data buffer
++ uint8_t* data = new uint8_t[len];
++ 
++ IPMI::completion_code cc = IPMI::CC_UNKBAD;
++ 
++ data[0] = 0x3;  
++ data[1] = 0x70;  
++ data[2] = 0x1;  
++ data[3] = 0x2;  //offset 0x2 for UIO info
++ l_err = IPMI::sendrecv(IPMI::master_readwrite(), cc, len, data);
```

Comment:
Should this instead be done the same way we do it for p9dsu, which is to keep the single type in Hostboot and do the 1u/2u query as part of the platform init in OPAL?

Seeing as we don't have a way to do per-platform code in Hostboot apart from bringing in patches, the approach of doing it in Skiboot would avoid carrying a hostboot patch forever.

Reply:
Who would need to do this change if it need to be done like p9dsu?

Reply:
I'd expect the patch to come from whoever is submitting the code.

Mostly I'm interested in it doing it the same way so there's some consistency and we have one fewer per platform patch sitting around for hostboot.

It probably isn't a blocker though for a p8 platform.